iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/statemachine/CreateRegionGroupsProcedure.java

```diff
@@ -173,40 +173,47 @@ protected Flow executeFromState(ConfigNodeProcedureEnv env, CreateRegionGroupsSt
         // Build RegionGroupCache immediately to make these successfully built RegionGroup available
         createRegionGroupsPlan
             .getRegionGroupMap()
+            .entrySet()
+            .parallelStream()
```

Comment:
I suggest we remove the parallelism here, as it's very unlikely that multiple sg regions will be created at the same time, but parallel here and below might introduce deadlock issues in some scenarios

Reply:
Fixed